### ISSUE TITLE
Give my tip for supporting iOS 8 and earlier

### ISSUE THREAD

**Issue by Khronoss** [phonegap-build/PushPlugin#332]
Hi,

I had some troubles since the last update of PushPlugin for iOS 8. In fact, there is the new method to register in APNs for iOS 8, but there is no call for the old method. Plus, the '#if' statement in PushPlugin.m to check if the iPhone is on iOS 8 or earlier is kinda...not working =X (at least for me).

So, here is my tip:
    I changed a bit the register:(CDVInvokedUrlCommand*)command that way

<code>\- (void)register:(CDVInvokedUrlCommand*)command; {

```
    self.callbackId = command.callbackId;

    NSMutableDictionary* options = [command.arguments objectAtIndex:0];

    UIRemoteNotificationType notificationTypes = UIRemoteNotificationTypeNone;

    id badgeArg = [options objectForKey:@"badge"];
    id soundArg = [options objectForKey:@"sound"];
    id alertArg = [options objectForKey:@"alert"];

if ([[[UIDevice currentDevice] systemVersion] compare:@"8.0" options:NSNumericSearch] != NSOrderedAscending) {
    UIUserNotificationType UserNotificationTypes = UIUserNotificationTypeNone;

    if ([badgeArg isKindOfClass:[NSString class]])
    {
        if ([badgeArg isEqualToString:@"true"]) {
            UserNotificationTypes |= UIUserNotificationTypeBadge;
        }
    }
    else if ([badgeArg boolValue]) {
        UserNotificationTypes |= UIUserNotificationTypeBadge;
    }

    if ([soundArg isKindOfClass:[NSString class]])
    {
        if ([soundArg isEqualToString:@"true"]) {
            UserNotificationTypes |= UIUserNotificationTypeSound;
        }
    }
    else if ([soundArg boolValue]) {
        UserNotificationTypes |= UIUserNotificationTypeSound;
    }

    if ([alertArg isKindOfClass:[NSString class]])
    {
        if ([alertArg isEqualToString:@"true"]) {
            UserNotificationTypes |= UIUserNotificationTypeAlert;
        }
    }
    else if ([alertArg boolValue]) {
        UserNotificationTypes |= UIUserNotificationTypeAlert;
    }

    notificationTypes |= UIRemoteNotificationTypeNewsstandContentAvailability;
    UserNotificationTypes |= UIUserNotificationActivationModeBackground;

    UIUserNotificationSettings *settings = [UIUserNotificationSettings settingsForTypes:UserNotificationTypes categories:nil];
    [[UIApplication sharedApplication] registerUserNotificationSettings:settings];
    [[UIApplication sharedApplication] registerForRemoteNotifications];
} else {
    if ([badgeArg isKindOfClass:[NSString class]])
    {
        if ([badgeArg isEqualToString:@"true"]) {
            notificationTypes |= UIRemoteNotificationTypeBadge;
        }
    }
    else if ([badgeArg boolValue]) {
        notificationTypes |= UIRemoteNotificationTypeBadge;
    }

    if ([soundArg isKindOfClass:[NSString class]])
    {
        if ([soundArg isEqualToString:@"true"]) {
            notificationTypes |= UIRemoteNotificationTypeSound;
        }
    }
    else if ([soundArg boolValue]) {
        notificationTypes |= UIRemoteNotificationTypeSound;
    }

    if ([alertArg isKindOfClass:[NSString class]])
    {
        if ([alertArg isEqualToString:@"true"]) {
            notificationTypes |= UIRemoteNotificationTypeAlert;
        }
    }
    else if ([alertArg boolValue]) {
        notificationTypes |= UIRemoteNotificationTypeAlert;
    }

    [[UIApplication sharedApplication] registerForRemoteNotificationTypes:notificationTypes];
}

self.callback = [options objectForKey:@"ecb"];

if (notificationTypes == UIRemoteNotificationTypeNone)
    NSLog(@"PushPlugin.register: Push notification type is set to none");

isInline = NO;

if (notificationMessage)            // if there is a pending startup notification
    [self notificationReceived];    // go ahead and process it
```

}</code>

Basically, you check the iOS version with <code>[[UIDevice currentDevice] systemVersion]</code> at the runtime. Then, if it's iOS 8 (or later e.g 9, 10....) you register with the iOS8 registering way, else you do it the old way.

Works fine for me :)
